ui/library/src/utils/scroll.ts

```diff
@@ -78,13 +71,60 @@ export function scrollToPosition(
     leftPixels = (width * (PDF_HEIGHT_POINTS - bottomPoints)) / PDF_HEIGHT_POINTS;
   }
 
-  targetDiv.scrollTo({
+  const pageId = generatePageIdFromIndex(pageIndex);
+
+  if (!pageId) {
+    return;
+  }
+
+  const pageIdElement = document.getElementById(pageId);
+
+  if (!pageIdElement) {
+    return;
+  }
+
+  const parentElement = pageIdElement.parentElement;
+
+  if (!parentElement) {
+    return;
+  }
+
+  if (
+    hasVerticalScrollbar(parentElement) &&
+    window.getComputedStyle(parentElement)['overflow'] === 'scroll'
+  ) {
+    parentElement.scrollTo({
+      top: Math.floor(heightWithMargins * pageIndex + marginTopPixels + (height - bottomPixels)),
+      left: Math.floor(leftPixels),
+      behavior: 'smooth',
+    });
+    return;
+  }
+
+  const outerParentElement = parentElement.parentElement;
+
+  if (!outerParentElement) {
+    return;
+  }
```

Comment:
There should be no need to handle multiple scrolling elements. If for some reason you _do_ want to handle it, then what you'd do is recursively find scrollable nodes up to the root and call [`node.scrollIntoView()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) on every node after the deepest one.